### DISCUSSION TITLE
Guia de boas práticas

### DISCUSSION THREAD

**Discussion by RodriAndreotti** [BrasilAPI/BrasilAPI#277] [Ideas]
Pessoal, estou abrindo a discussão só para tentarmos chegar a melhor solução, abri este o PR #268 com algumas, que considero, boas práticas e alguns formatos de retorno que podem facilitar o consumo do API.

No PR fiz a sugestão do uso dos endpoints em inglês pensando em uma padronização e também baseado num comentário do @lucianopf  em outro PR de algum tempo atrás.

O @murilohns notou que hoje temos somente uma rota com o endpoint em inglês (banks), o restante está no formato de siglas em português (ddd, cep, uf etc) ou em português (feriado).

O que vocês acham sobre a questão dos endpoints? Aliás, seria interessante receber também sugestões sobre os demais itens... :-P

Acabei demorando para abrir essa discussão, pois estou com dois projetos com deadline meio apertado e está punk achar tempo... kkk

**Comment by pedrosancao** [comment]
Como as siglas são de palavras em português eu acho que fica mais consistente que os endpoints também sejam em português, acho que as outra opções ficam estranhas:
- siglas de palavras em português e palavras em inglês
- traduzir siglas e palavras para inglês

**Reply by RodriAndreotti** [reply]
Concordo, contigo, eu pensei bastante sobre isso, e como o foco é transformar o Brasil em um API faz sentido que os endpoints sejam em português, agora a dúvida.
Se inserirmos essa recomendação, seria legal trocar o endpoint de bancos (banks) para português? (Correndo o risco de quebrar para alguém que já esteja usando)

Ou será que existiria uma forma de fazer um redirect nas requisições a este endpoint para um novo e evitar que quebre?

**Reply by pedrosancao** [reply]
Acho que é uma boa estratégia renomear o endpoint e adicionar um redirecionamento com status 301 _Moved Permanently_ no termo em inglês, não sei que a Vercel fornece algum relatório de uso, para do futuro descomissionar o endpoint com nome em inglês.

**Reply by RodriAndreotti** [reply]
Boa, seria interessante mesmo.
Mudando de pato para ganso, o que vc acha de incluirmos algumas boas práticas de código também?
Exemplo, venho do PHP, e existe os PSR-1 e 2 que sugerem algumas boas práticas como:
- camelCase para variáveis, métodos e funções
- PascalCase para classes
- Chaves na próxima linha para classes, métodos e funções
E outras sugestões.

Na sua opinião, seria interessante termos algo para o projeto?

**Reply by pedrosancao** [reply]
já tem o ESLint configurado para padronizar e acho que já faz essa verificação ao fazer commit, eu apenas mencionaria o uso do ESLint.
